src/pages/blog/[slug]/opengraph-image.ts

```diff
@@ -0,0 +1,37 @@
+import { getCollection } from "astro:content";
+import { getSlug, loadFont, Template } from "../../../utils";
+import type { APIRoute } from "astro";
+import { GenerateImageResponse } from "../../../lib/astro-opengraph-image";
+
+export const prerender = true;
+
+export async function getStaticPaths() {
+  const blogEntries = await getCollection("blog");
+
+  return blogEntries
+    .filter((post: any) => post.data.isDraft !== true)
```

Comment:
why using any

Reply:
@suyash5641 I believe by using the command `astro sync`, you should get the type inference for the blog collection.

https://docs.astro.build/en/reference/cli-reference/#astro-sync

Reply:
@yaralahruthik Were you referring to this way  

import type { CollectionEntry } from "astro:content";
type Blog = CollectionEntry<"blog">;

and use like this 

 .filter((post: Blog) => !post.data.isDraft)
    .map((post: Blog) => ({
      params: { slug: getSlug(post.data.title) },
      props: { post },
    }));

Reply:
@suyash5641 I mean when you get the data with `getCollection` it automatically infers the type from the zod schema. So, no explicit typing is required. Like if you just remove `any`, it should work. But given how astro works, you might have to run the `astro sync` or restart the `TS server` to get astro to infer the types properly.

`const allBlogPosts = await getCollection("blog");`